nomad/consul_oss_test.go

```diff
@@ -0,0 +1,123 @@
+//+build !ent
+
+package nomad
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/hashicorp/nomad/command/agent/consul"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConsulACLsAPI_CheckPermissions_oss(t *testing.T) {
+
+	// In Nomad OSS, CheckPermissions will only receive "" as input for the
+	// namespace parameter - as the ConsulUsage map from namespace to usages will
+	// always contain one key - the empty string.
+
+	t.Parallel()
+
+	try := func(t *testing.T, namespace string, usage *structs.ConsulUsage, secretID string, exp error) {
+		logger := testlog.HCLogger(t)
+		aclAPI := consul.NewMockACLsAPI(logger)
+		cAPI := NewConsulACLsAPI(aclAPI, logger, nil)
+
+		err := cAPI.CheckPermissions(context.Background(), namespace, usage, secretID)
+		if exp == nil {
+			require.NoError(t, err)
+		} else {
+			require.Equal(t, exp.Error(), err.Error())
+		}
+	}
+
+	t.Run("check-permissions kv read", func(t *testing.T) {
+		t.Run("uses kv has permission", func(t *testing.T) {
```

Comment:
I don't know how I personally feel about this style of nested `t.Run`: When the assertions are simply delegation to `try`, it feels a simple table tests might read better and is more concise.

Reply:
Yeah, for tests like these it's hard to find the right balance between readability of the code and readability of the _description_ of the test. By composing `t.Run` we get a free way of composing the description, making it obvious what the difference is between one particular test from the tests surrounding it. OTOH it's a lot of lines of code that aren't actually doing anything. I'll take a look at refactoring these tests some more after the bugfix release.